tests/integrations/test_crossed_integrations.py

```diff
@@ -29,14 +29,25 @@ class Test_PythonKafka:
     @staticmethod
     def get_span(interface, span_kind, topic):
 
-        logger.debug(f"Try to found traces with span kind: {span_kind} and topic: {topic} in {interface}")
+        logger.debug(f"Trying to find traces with span kind: {span_kind} and topic: {topic} in {interface}")
 
         for data, trace in interface.get_traces():
             for span in trace:
                 if span_kind != span["meta"].get("span.kind"):
                     continue
 
-                if topic != span["meta"].get("kafka.topic"):
+                is_java_tracer = "java" in span["meta"]["component"]
+
+                # dd-trace-java does not add kafka.topic to its kafka.produce spans
+                if not is_java_tracer and topic != span["meta"].get("kafka.topic"):
+                    continue
+
+                # instead, we rely on resource
+                if is_java_tracer and not span["resource"].endswith(topic):
+                    continue
+                if ("java" not in span["meta"]["component"] and topic != span["meta"].get("kafka.topic")) or (
```

Comment:
Well, then this part is not needed anymore (it's splitted in the two `if` above)